File: Problems/Algorithms/844. Backspace String Compare/backspace_string_compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        return self.process(s) == self.process(t)

    def process(self, s: str) -> str:
        stack = []

        for ch in s:
            if ch == '#':
                if stack:
                    stack.pop()
                else:
                    continue
            else:
                stack.append(ch)

        return ''.join(stack)

File: Problems/Algorithms/844. Backspace String Compare/backspace_string_compare.rs
impl Solution {
    pub fn backspace_compare(s: String, t: String) -> bool {
        fn process(s: String) -> Vec<char> {
            let mut stack = vec![];
            for ch in s.chars() {
                match ch {
                    '#' => {
                        if !stack.is_empty() {
                            stack.pop();
                        }
                    },
                    _ => stack.push(ch),
                }
            }
            stack
        }
        
        process(s) == process(t)
    }
}

